### ISSUE TITLE
[BUG] python 3.12 compatibility

### ISSUE THREAD

**Issue by ml31415** [hadialqattan/pycln#219]
Python 3.12 has removed the distutils package. For that reason, trying to install `pycln` runs into this issue:

```
 Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repo9lp39qm0/py_env-python3/bin/pycln", line 5, in <module>
    from pycln.cli import app
  File "/home/runner/.cache/pre-commit/repo9lp39qm0/py_env-python3/lib/python3.12/site-packages/pycln/cli.py", line 8, in <module>
    from .utils import iou, pathu, refactor, regexu, report
  File "/home/runner/.cache/pre-commit/repo9lp39qm0/py_env-python3/lib/python3.12/site-packages/pycln/utils/pathu.py", line 4, in <module>
    from distutils import sysconfig
ModuleNotFoundError: No module named 'distutils'
Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repo9lp39qm0/py_env-python3/bin/pycln", line 5, in <module>
    from pycln.cli import app
  File "/home/runner/.cache/pre-commit/repo9lp39qm0/py_env-python3/lib/python3.12/site-packages/pycln/cli.py", line 8, in <module>
    from .utils import iou, pathu, refactor, regexu, report
  File "/home/runner/.cache/pre-commit/repo9lp39qm0/py_env-python3/lib/python3.12/site-packages/pycln/utils/pathu.py", line 4, in <module>
    from distutils import sysconfig
ModuleNotFoundError: No module named 'distutils'
```

`distutils` is deprecated since 3.10, so it's probably about time to replace it. As `pycln` is frequently used in pre-commit hooks, this unfortunately blocks packages from adding support for 3.12 themself or otherwise forces them to ditch `pycln`.

**Comment by hadialqattan** [comment]
Hi @ml31415, I've just merged a PR #220 that contains the fix.

**Comment by hadialqattan** [comment]
I'll publish a release within few minutes.

**Comment by hadialqattan** [comment]
v2.3.0 has been just released!

**Comment by hadialqattan** [comment]
Thanks for your supports @ml31415 .

**Comment by ml31415** [comment]
Thanks for your quick response. Unfortunately, it looks like that was only half of the story of py3.12 compatibility. That's the next issue popping up on a 3.12 install:
```
pycln (Python unused imports)............................................Failed
- hook id: pycln
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repoiqxf7xrm/py_env-python3/bin/pycln", line 5, in <module>
    from pycln.cli import app
  File "/home/runner/.cache/pre-commit/repoiqxf7xrm/py_env-python3/lib/python3.12/site-packages/pycln/cli.py", line 8, in <module>
    from .utils import iou, pathu, refactor, regexu, report
  File "/home/runner/.cache/pre-commit/repoiqxf7xrm/py_env-python3/lib/python3.12/site-packages/pycln/utils/refactor.py", line 5, in <module>
    from pathlib import Path, _posix_flavour, _windows_flavour  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name '_posix_flavour' from 'pathlib' (/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/pathlib.py)
Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repoiqxf7xrm/py_env-python3/bin/pycln", line 5, in <module>
    from pycln.cli import app
  File "/home/runner/.cache/pre-commit/repoiqxf7xrm/py_env-python3/lib/python3.12/site-packages/pycln/cli.py", line 8, in <module>
    from .utils import iou, pathu, refactor, regexu, report
  File "/home/runner/.cache/pre-commit/repoiqxf7xrm/py_env-python3/lib/python3.12/site-packages/pycln/utils/refactor.py", line 5, in <module>
    from pathlib import Path, _posix_flavour, _windows_flavour  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name '_posix_flavour' from 'pathlib' (/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/pathlib.py)
```
I'd recommend to add py3.11 and py3.12 to your test matrix in github actions: https://github.com/hadialqattan/pycln/blob/e15df2ee59593704ef85262ee22aeb447aeb3b20/.github/workflows/ci.yml#L40
        
Also in the fuzzing config: https://github.com/hadialqattan/pycln/blob/e15df2ee59593704ef85262ee22aeb447aeb3b20/.github/workflows/fuzz.yml#L16

**Comment by ml31415** [comment]
My suggestion for a fix of: https://github.com/hadialqattan/pycln/blob/e15df2ee59593704ef85262ee22aeb447aeb3b20/pycln/utils/refactor.py#L30

Instead of subclassing `pathlib.Path`, just use `is_stub_file` on it directly. I'd be surprised, if there is an actual speed concern, that would justify the hazzle with caching etc. And surely nothing to justify accessing private implementation details of `pathlib`. Then there is also no need for a disambiguation of win or unix paths anymore, as `pathlib.Path` does this very well.

**Comment by hadialqattan** [comment]
This isssue has been resolved in PR #224 .. Thanks to @perfa

**Comment by hadialqattan** [comment]
@ml31415 
Please checkout v2.4.0
